<launch>

    <arg name="record_bag" default="true" />
    <arg name="start_time" default="undef"/>
    <arg name="world_path" default="$(find local_planner)/../sim/worlds/simple_obstacle.world" />

    <include file="$(find local_planner)/launch/local_planner_sitl_mavros.launch" >
        <arg name="world_path" value="$(arg world_path)" />
        <arg name="point_cloud_topic" value="/camera/depth/points" />
    </include>

     <node name="local_planner_node" pkg="local_planner" type="local_planner_node" output="screen">
        <param name="goal_x_param" value="30" />
        <param name="goal_y_param" value="0"/>
        <param name="goal_z_param" value="4" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find local_planner)/resource/local_planner.rviz" />

    <!-- Logging -->
    <node pkg="rosbag" type="record" name="log_bag" args="record -O log_$(arg start_time) /mavros/local_position/pose /mavros/local_position/velocity /goal_position /camera/depth/points /local_pointcloud /blocked_marker /rejected_marker /candidates_marker /selected_marker /current_setpoint --split --size 500 --max-splits 5" if="$(arg record_bag)" />

</launch>
